Sources/Rendering/OpenGL/Texture/index.js

```diff
@@ -1120,17 +1135,149 @@ function vtkOpenGLTexture(publicAPI, model) {
 
     const haveWebgl2 = model.openGLRenderWindow.getWebgl2();
 
+    let reformatGradientsFunction;
+    if (haveWebgl2) {
+      reformatGradientsFunction = (workerResults) => {
+        const numVoxelsIn = width * height * depth;
+        const reformattedGradients = new Uint8Array(numVoxelsIn * 4);
+        const maxMag = model.volumeInfo.max;
+
+        workerResults.forEach(
+          ({
+            subGradients,
+            subMagnitudes,
+            subMinMag,
+            subMaxMag,
+            subDepthStart,
+            subDepthEnd,
+          }) => {
+            let inIdx = 0;
+            let inMagIdx = 0;
+            let outIdx = subDepthStart * width * height * 4;
+            // start and end depths are inclusive
+            const numWorkerVoxels =
+              width * height * (subDepthEnd - subDepthStart + 1);
+            for (let vp = 0; vp < numWorkerVoxels; ++vp) {
+              reformattedGradients[outIdx++] = subGradients[inIdx++];
+              reformattedGradients[outIdx++] = subGradients[inIdx++];
+              reformattedGradients[outIdx++] = subGradients[inIdx++];
+              reformattedGradients[outIdx++] =
+                255.0 * Math.sqrt(subMagnitudes[inMagIdx++] / maxMag);
+            }
+          }
+        );
+
+        return publicAPI.create3DFromRaw(
+          width,
+          height,
+          depth,
+          4,
+          VtkDataTypes.UNSIGNED_CHAR,
+          reformattedGradients
+        );
+      };
+    } else {
+      // Now determine the texture parameters using the arguments.
+      publicAPI.getOpenGLDataType(VtkDataTypes.UNSIGNED_CHAR);
+      publicAPI.getInternalFormat(VtkDataTypes.UNSIGNED_CHAR, 4);
+      publicAPI.getFormat(VtkDataTypes.UNSIGNED_CHAR, 4);
```

Comment:
These functions all are called `get...` but they mutate internal state, which is a bit confusing. Might consider renaming them in the next major version.

Reply:
maybe that was a convention imported from c++? 
cc: @martinken @jourdain

Reply:
I'm not sure what those methods do, but I agree `get` should not mutate the state...

Reply:
Referenced here: https://github.com/Kitware/vtk-js/issues/639

Reply:
It is exactly because someone wrote them that way in C++. I am not a fan of how it is done either.